Updates to DevRev Destination - Contacts (RevUsers) no longer require accounts and tags are properly applied

RtpChris opened the pull request:
<!-- Hello and thank you for contributing to Segment action-destinations! -->

<!-- Before opening your pull request, make sure you have added and ran unit
     tests and tested your change locally. Refer to our testing
     documentation for more information: https://github.com/segmentio/action-destinations/blob/main/docs/testing.md -->

<!-- If you have questions or issues please open a new issue or create a new discussion
     post in Github. -->

This pull request fixes one major issue, and updates Contact creation to match our updated policies

1) In the past, it was required to have an account for a Contact (RevUser), as this is no longer required the code has been updated to support this.
2) Tags set in CreateRevUser were only getting set when the Contact was created, not when there was an existing Contact.  Now the tag is applied in both cases.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

joe-ayoub-segment commented on packages/destination-actions/src/destinations/devrev/createRevUser/index.ts:
```diff
@@ -124,65 +123,70 @@ const action: ActionDefinition<Settings, Payload> = {
+      let requestUrl = ''
+      //Search using domain if the email domain is not blacklisted
+      // Otherwise search the external refs
+      if (domain !== email) requestUrl = `${getBaseUrl(settings)}${devrevApiPaths.accountsList}?domains="${domain}"`
```
hi @RtpChris , 
Why would domain ever be equal to an email address?

joe-ayoub-segment commented:
hi @RtpChris thanks for raising this PR. 

Could you please run `yarn types` and commit the result?

Would you be able to provide evidence of testing to demonstrate that the new code works as expected and doesn't break the experience for customers who are already using the I please?

RtpChris commented:
Hey @joe-ayoub-segment - I've updated the types.

I've also attached examples of the testing I've done

![API Calls](https://github.com/user-attachments/assets/89df0559-3b74-4716-9ccc-57cf9eaa4a9e)

![Blacklisted Email Domain](https://github.com/user-attachments/assets/16c22e2f-eb9d-485b-b461-0cdf9d74f01b)

![Non-blacklisted email domain](https://github.com/user-attachments/assets/844c52a2-bee8-4fa8-b63d-280b336c5c5b)

While going through and documenting this, I discovered a corner case I originally did not include proper handling for this change (where the email has a domain that is not blacklisted and there is no existing account).  I'm adding that now and will add additional coverage for that case shortly.

RtpChris commented:
Ok I've updated the code and updated the tests - as part of that I realized that I had removed checks in the tests that should have been there.  I've added those back and just left the one test that should have been affected by this change modified.  The other changes to the tests were typos/reordering checks to make sense.

RtpChris commented:
Here's the screenshots of the testing:

1) New RevUser, existing account. We can see that the new RevUser was created under the existing account
![Screenshot 2024-07-16 180308](https://github.com/user-attachments/assets/5a35c112-6f72-44b1-a414-4e9602010f06)
![Screenshot 2024-07-16 180407](https://github.com/user-attachments/assets/3a15550b-2980-43c4-b722-f6efd2da24ed)

2) New RevUser, New Account. We can see that since the account wasn't found and the domain wasn't blacklisted, we created the new account properly

![Screenshot 2024-07-16 180437](https://github.com/user-attachments/assets/572d5358-e7d0-4aa8-a8c9-2edb9ecf7f07)

![Screenshot 2024-07-16 180510](https://github.com/user-attachments/assets/cbc46f52-d120-41ab-8b2f-50858cdbb286)

RtpChris commented:
Also, more as an FYI, when I ran `yarn types`  it generated a types file in `liveramp-audiences`; I didn't include that file here since it's not my destination but figured it would be good to know about.

joe-ayoub-segment commented:
hi @RtpChris PR deployed.
